components/channel_view/index.ts

```diff
@@ -20,20 +20,29 @@ import {TutorialSteps, Preferences} from 'utils/constants';
 import {goToLastViewedChannel} from 'actions/views/channel';
 import {setShowNextStepsView} from 'actions/views/next_steps';
 import {isOnboardingHidden, showNextSteps, showNextStepsTips} from 'components/next_steps_view/steps';
+import {Action, ActionFunc, GenericAction} from 'mattermost-redux/types/actions';
+import {GlobalState} from 'types/store';
+import {UserProfile} from 'mattermost-redux/types/users';
 
 import ChannelView from './channel_view';
 
+type Actions = {
+    goToLastViewedChannel: () => Promise<{data: boolean}>;
+    setShowNextStepsView: (show: boolean) => Action;
+    getProfiles: (page?: number, perPage?: number, options?: any) => ActionFunc;
+}
+
 // Temporary selector until getDirectTeammate is converted to be redux-friendly
 const getDeactivatedChannel = createSelector(
-    (state: any, channelId: any) => {
+    (state: GlobalState, channelId: string) => {
         return getDirectTeammate(state, channelId);
     },
     (teammate: any) => {
```

Comment:
I wasn't sure how to change this one to be not any. I think this is supposed to be UserProfile but it give me this error:

`No overload matches this call.
  The last overload gave the following error.
    Argument of type '(state: GlobalState, channelId: string) => {}' is not assignable to parameter of type 'ParametricSelector<unknown, unknown, UserProfile>[]'.
      Type '(state: GlobalState, channelId: string) => {}' is missing the following properties from type 'ParametricSelector<unknown, unknown, UserProfile>[]': pop, push, concat, join, and 27 more.ts(2769)`

If you are supposed to change this let me know!

Reply:
Have you tried doing `(state: GlobalState, channelId: string) => getDirectTeammate(state, channelId)` ?

Reply:
Probably I`d change that any for `(teammate: Record<string, any>) => {` until the `utils.jsx` is moved to TS